File: frame和bounds/frame和bounds/ViewController.m
//
//  ViewController.m
//  frame和bounds
//
//  Created by Jerod on 2021/4/2.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor whiteColor];
    
    UIView *view1 = [[UIView alloc] initWithFrame:CGRectMake(50, 50, 200, 200)];
    [view1 setBounds:CGRectMake(-30, -30, 100, 100)];
    view1.backgroundColor = [UIColor systemPinkColor];
    [self.view addSubview:view1];
    NSLog(@"view1 frame:%@========view1 bounds:%@",NSStringFromCGRect(view1.frame),NSStringFromCGRect(view1.bounds));
    
    UIView *view2 = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    view2.backgroundColor = [[UIColor brownColor] colorWithAlphaComponent:0.5];
    [view1 addSubview:view2];//添加到view1上,[此时view1坐标系左上角起点为(-30,-30)]
    NSLog(@"view2 frame:%@========view2 bounds:%@",NSStringFromCGRect(view2.frame),NSStringFromCGRect(view2.bounds));
    
    
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapAction)];
    [view1 addGestureRecognizer:tap];
}

- (void)tapAction {
    NSLog(@"%s", __func__);
}

@end

File: VerifyNum/VerifyNum/ViewController.swift
//
//  ViewController.swift
//  VerifyNum
//
//  Created by Jerod on 2021/3/6.
//

import UIKit

class ViewController: UIViewController, NumChineseDialogDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.view.backgroundColor = .white
        
        let btn = UIButton(type: .custom)
        btn.setTitle("开始验证", for: .normal)
        btn.backgroundColor = .systemPink
        btn.addTarget(self, action: #selector(btnClick(_:)), for: .touchUpInside)
        btn.frame = CGRect(x: 10, y: 100, width: 100, height: 50)
        self.view.addSubview(btn)
        
        
    }

    @objc func btnClick(_ sender:UIButton) {
         
        let dialog = NumChineseDialog(verify: 4, show: 6)
        dialog.delegate = self
        dialog.show()
        
    }

    func verifySuccess() {
        NSLog("bingo")
    }
    
    func verifyFailure(clickArr: [String]) {
        NSLog("error %@", clickArr)
    }
    
//    func verifyChineseButtonClicked(clkickArr: [String], chinese: String, selected: Bool) {
//
//    }
    
}

